File: LC36.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        seen = set()
        for i in range(len(board)):
            for j in range(len(board[i])):
                char = board[i][j]
                if (char != '.'):
                    row = char + " found in row " + str(i)
                    col = char + " found in col " + str(j)
                    box = char + " found in box " + str(i//3) + "-" +  str(j//3)
                    if (row in seen or col in seen or box in seen):
                        return False
                    seen.add(row)
                    seen.add(col)
                    seen.add(box)
        return True

File: LC36.java
import java.util.HashSet;

public class LC36 {
    public boolean isValidSudoku(char[][] board){
        HashSet<String> set = new HashSet<>();
        for (int i = 0; i < board.length; i++){
            for (int j = 0; j < board[i].length; j++){
                char current_val = board[i][j];
                if (current_val != '.'){
                    if(!set.add(current_val + " found in row " + i) ||
                    !set.add(current_val + " found in column  " + j) ||
                    !set.add(current_val + " found in box " + i/3 + "-" + j/3)) return false; 
                }
            }
        }
        return true;
    }
}
